srcopsmetrics/github_handling.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python3
+# Meta-information Indicators
+# Copyright(C) 2021 Dominik Tuchyna
+#
+# This program is free software: you can redistribute it and / or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Module that contains handling decorator for GitHub API Rate limit."""
+
+import logging
+import os
+import time
+from datetime import datetime, timezone
+from typing import Optional
+
+from github import Github
+
+_LOGGER = logging.getLogger(__name__)
+
+_GITHUB_ACCESS_TOKEN = os.getenv("GITHUB_ACCESS_TOKEN")
+
+API_RATE_MINIMAL_REMAINING = 80
+GITHUB_TIMEOUT_SECONDS = 60
+
+
+class GithubHandler:
+    """Handler class that contains GH API rate handling logic."""
+
+    def __init__(self, github: Optional[Github] = None):
+        """Initialize with github object."""
+        if not github:
+            github = Github(login_or_token=_GITHUB_ACCESS_TOKEN, timeout=GITHUB_TIMEOUT_SECONDS)
+
+        self.github = github
+        self.remaining = github.get_rate_limit().core.remaining
+
+    def _is_api_exhausted(self):
+        """Check if GH API rate limit is exhausted."""
+        self.remaining = self.github.get_rate_limit().core.remaining
```

Comment:
isnt this already set on the line 45 ?

Reply:
it is, but this is just assigning it everytime we check. Without get_rate_limit() you don't really know about the _current_ remaining limit. This is then used in `iterator`

Reply:
ah I see, how this is being used :+1: